config/watcher.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package config
+
+import (
+	"context"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/reload"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+type reloader struct {
```

Comment:
Where is it being instanciated? It feels like black magic

Reply:
https://github.com/elastic/cloudbeat/blob/5d858d58c6b9d6750918f7ad22b9e3df226ad752/config/updates.go#L56